### ISSUE TITLE
Sprockets::FileNotFound in Calagator/site#index

### ISSUE THREAD

**Issue by shawnacscott** [calagator/calagator#350]
I tried to follow the [new engine setup instructions](https://github.com/calagator/calagator/commit/0962ccc144563cbc94a60aa454fe8252215d51b2) in a new Rails app, and on starting the server, got the error:
````
Sprockets::FileNotFound in Calagator/site#index

couldn't find file 'leaflet.awesome-markers'
````
Here's the [full error output and trace](https://gist.github.com/shawnacscott/ffb2e992bb8f56ad84f9).

Steps I took to get to this point:

* `rails _3.2.21_ new <app>` (I believe it must be version 3.2.21, since gem currently only supports 3.2 pending #351)
* `cd <app>`
* Add `gem 'calagator'` to Gemfile
* `bundle install`
* `bundle exec rails g calagator:install`
* `bundle exec rails s`

**Comment by reidab** [comment]
I think that's in Calagator's vendor/assets folder, which might need special configuration to work with the engine.

**Comment by botandrose** [comment]
Interesting, seems that vendor assets aren't shared in gems. Seems
straightforward enough to move these assets to somewhere that's in the
sprockets load path, but I wonder why the tests aren't catching this? A
failing test would be nice.

**Comment by reidab** [comment]
We can also just add vendor/assets to the assets paths provided by the engine, which I like better than moving these into app/assets.

**Comment by edrex** [comment]
this no longer seems to be happening using the latest calagator@master

**Comment by reidab** [comment]
http://stackoverflow.com/questions/10224296/add-asset-path-in-rails-mountable-engine

**Comment by shawnacscott** [comment]
I'm also no longer seeing this error. I'll try a new checkout soon and see if I can reproduce, but this may be fixed.

**Comment by shawnacscott** [comment]
Ok, I tried this again with a new checkout and compared the pre-release gem and the current master branch, and current master seems not to have this problem. Assuming there's not something strange happening int he packaging process for gems that isn't surfacing, I'm going to close this for now unless there's another reproduction case/someone says otherwise.

**Comment by antislice** [comment]
I know this is closed, but in interest of not duplicating bugs I figured I'd leave a comment.

I got a very similar error when trying to set up calagator with a new host application. I worked around the lack of a calagator executable by running `calagator new wwcalagator` inside my clone of calagator (master branch), then copying the resulting app to a different directory outside calagator.

The error is below, but basically my app couldn't find leaflet.awesome-markers (referenced in calagator.js & index.scss) locally and threw up. I worked around it by adding leaflet.awesome-markers.js in my app's vendor/assets.

Setting up the dummy app did not have this issue.

ActionView::Template::Error (couldn't find file 'leaflet.awesome-markers' with type 'application/javascript'):
    25:     
    26:     <%= yield :css_insert %>
    27: 
    28:     
    29:     <%= javascript_include_tag *mapping_js_includes %>
    30:     <%= javascript_include_tag 'application' %>
    31: 
  calagator (1.0.0.rc2) app/assets/javascripts/calagator.js:4
  sprockets (3.2.0) lib/sprockets/resolve.rb:64:in `resolve!'
  sprockets (3.2.0) lib/sprockets/directive_processor.rb:399:in `resolve'
  sprockets (3.2.0) lib/sprockets/directive_processor.rb:207:in `process_require_directive'
  sprockets (3.2.0) lib/sprockets/directive_processor.rb:180:in `block in process_directives'
  sprockets (3.2.0) lib/sprockets/directive_processor.rb:178:in `each'
  sprockets (3.2.0) lib/sprockets/directive_processor.rb:178:in `process_directives'
  sprockets (3.2.0) lib/sprockets/directive_processor.rb:83:in `_call'
  sprockets (3.2.0) lib/sprockets/directive_processor.rb:68:in `call'
  sprockets (3.2.0) lib/sprockets/processor_utils.rb:75:in `call_processor'
  sprockets (3.2.0) lib/sprockets/processor_utils.rb:57:in `block in call_processors'
  sprockets (3.2.0) lib/sprockets/processor_utils.rb:56:in `reverse_each'
  sprockets (3.2.0) lib/sprockets/processor_utils.rb:56:in `call_processors'
  sprockets (3.2.0) lib/sprockets/loader.rb:86:in `load_asset_by_uri'
  sprockets (3.2.0) lib/sprockets/loader.rb:45:in `block in load'
  sprockets (3.2.0) lib/sprockets/loader.rb:155:in `fetch_asset_from_dependency_cache'
  sprockets (3.2.0) lib/sprockets/loader.rb:38:in `load'
  sprockets (3.2.0) lib/sprockets/cached_environment.rb:20:in `block in initialize'
